.github/workflows/ci.yml

```diff
@@ -0,0 +1,125 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+      - staging
+      - trying
+  pull_request:
+    branches:
+      - master
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  check:
+    name: Check
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+    steps:
+      - uses: actions/checkout@v2
+      - run: cargo check --all-targets
+
+  test:
+    name: Test
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Test
+        run: |
+          cargo test --lib -- --skip test_vm
+          cargo test --bins --examples
+          cargo test --doc
+
+  integration-tests:
+    name: Integration Tests
+    runs-on: [self-hosted]
+    steps:
+      - name: Install gdb
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y --no-install-recommends gdb
+      - uses: actions/checkout@v2
+        with:
+          lfs: true
+      - run: git lfs checkout benches_data/hello_world
+      - name: Integration Tests
+        run: cargo test --lib --test '*'
+
+  rusty_demo:
+    name: rusty_demo
+    runs-on: [self-hosted]
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          path: uhyve
+      - uses: actions/checkout@v2
+        with:
+          repository: hermitcore/rusty-hermit
+          path: rusty-hermit
+          submodules: true
+      - name: Build rusty_demo
+        run: cargo build -p rusty_demo
+        working-directory: rusty-hermit
+      - name: Run rusty_demo
+        run: RUST_LOG=debug cargo run -- --verbose ../rusty-hermit/target/x86_64-unknown-hermit/debug/rusty_demo
+        working-directory: uhyve
+      - name: Run rusty_demo
+        run: RUST_LOG=debug cargo run --release -- --verbose ../rusty-hermit/target/x86_64-unknown-hermit/debug/rusty_demo
+        working-directory: uhyve
+
+  fmt:
+    name: Format
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Format
+        run: cargo fmt --all -- --check
+      - name: Format Test Kernels
+        run: cargo fmt --all -- --check
+        working-directory: tests/test-kernels
+
+  clippy:
+    name: Clippy
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
```

Comment:
Do we need to run clippy on macos?

Reply:
Yes. We'll want to run clippy with all targets to catch all conditionally compiled code. Running it natively is easiest.